src/System.Runtime.Extensions/src/System/Environment.Unix.cs

```diff
@@ -356,6 +356,13 @@ private static int FindAndParseNextNumber(string text, ref int pos)
                 }
                 else break;
             }
+
+            // Integer overflow can occur for example with:
+            //     Linux nelknet 4.15.0-24201807041620-generic
+            // To form a valid Version, num must be positive.
+            if (num < 0)
+                num = int.MaxValue;
```

Comment:
Should we clamp and bail as soon as it overflows? Otherwise it can wrap back around to a truncated positive value, right?

Reply:
Good point.